src/slam_karto.cpp

```diff
@@ -160,6 +160,125 @@ SlamKarto::SlamKarto() :
   mapper_ = new karto::Mapper();
   dataset_ = new karto::Dataset();
 
+  // Setting General Parameters from the Parameter Server
+  bool use_scan_matching;
+  if(private_nh_.getParam("use_scan_matching", use_scan_matching) != mapper_->getParamUseScanMatching())
```

Comment:
this isn't quite doing what you think it is -- NodeHandle.getParam returns a bool regardless of the type of the data. The bool is true if the parameter existed on the server, false otherwise. Thus, the != X part of this should be removed.